mathesar/views/api.py

```diff
@@ -292,3 +301,50 @@ def create(self, request):
         serializer = DataFileSerializer(data=request.data, context={'request': request})
         serializer.is_valid(raise_exception=True)
         return create_datafile(request, serializer.validated_data)
+
+
+class ConstraintViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixin, CreateModelMixin, DestroyModelMixin):
+    serializer_class = ConstraintSerializer
+    pagination_class = DefaultLimitOffsetPagination
+
+    def get_queryset(self):
+        return Constraint.objects.filter(table__id=self.kwargs['table_pk']).order_by('-created_at')
+
+    def create(self, request, table_pk=None):
+        table = get_table_or_404(table_pk)
+        serializer = ConstraintSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        # If we don't do this, the request.data QueryDict will only return the last column's name
+        # if there are multiple columns.
+        data = request.data.dict()
+        data['columns'] = request.data.getlist('columns')
+        try:
+            constraint = table.add_constraint(data)
+        except ProgrammingError as e:
+            if type(e.orig) == DuplicateTable:
+                raise ValidationError(
+                    'Constraint with the same name already exists'
```

Comment:
This will also happen if a table with the desired name already exists (or any Relation).  I suggest just using the generic "Relation" to avoid confusion.